### PR TITLE
refactor(core-transactions): unique ipfs hashes

### LINKED ISSUE #3203: [2.6][T3] global ipfs hash uniqueness is not ensured
It is currently possible to register the same ipfs hash on the blockchain from different wallets. While this is clearly marked as a TODO in the source code it is still an issue.

https://github.com/ArkEcosystem/core/blob/9aaf1053c8141fc0ce529e9bc26869e84e4ec226/packages/core-transactions/src/handlers/ipfs.ts#L42-L53

This could be easily resolved by adding a new wallet index.

### PR DESCRIPTION
**Pull request by dated**
## Summary

Closes #3203.

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: packages/core-transactions/src/handlers/ipfs.ts [hunk 1]
```diff
@@ -44,8 +45,9 @@ export class IpfsTransactionHandler extends TransactionHandler {
         wallet: State.IWallet,
         walletManager: State.IWalletManager,
     ): Promise<void> {
-        // TODO implement unique ipfs hash on blockchain (not just on wallet)
-        if (wallet.hasAttribute("ipfs") && wallet.getAttribute("ipfs.hashes")[transaction.data.asset.ipfs]) {
+        const ipfsHashes = walletManager.getIndex("ipfs").keys();
```
Comment (supaiku0) [inline packages/core-transactions/src/handlers/ipfs.ts]: You can use `walletManager.getIndex("ipfs").has(value)` then the lookup is O(1).

Comment (dated) [reply packages/core-transactions/src/handlers/ipfs.ts]: Thanks!

**Comment by vasild** [comment]
This probably is the reason why devnet wouldn't sync (develop branch at 89fc75762):
```
[2019-11-12 15:03:15.108] ERROR: Failed to apply all transactions in block - reverting previous transactions
[2019-11-12 15:03:15.109] WARN : Refused new block {"version":0,"timestamp":82871848,"height":4068953,"previousBlockHex":"86866b974cc6ae5c86f145a810db927d81224d0f8b9e3c12ad2ceaf4902a964b","previousBlock":"86866b974cc6ae5c86f145a810db927d81224d0f8b9e3c12ad2ceaf4902a964b","numberOfTransactions":1,"totalAmount":"0","totalFee":"500000000","reward":"200000000","payloadLength":32,"payloadHash":"7f8a7db8a0cf5a8781856820cdad278b9c1a438e7975b1c0f9684106d2663e77","generatorPublicKey":"03f6af8c750b9d29d9da3d4ddf5818a1fcdd4558ba0dde731f9c4b17bcbdcd83f2","blockSignature":"304402207a4dce6d076a0642a7b4273b0b7dc6079d3027053727fd18ac62c6930503cfb002201d62a5fe21c5f6204fc7029bd93318710b8576709376ec4679ed9b748bf1e524","idHex":"73ffdb2d06f00666c66cc11bcecd4800228351e4db50cc5e7a3641fc6dc91f5a","id":"73ffdb2d06f00666c66cc11bcecd4800228351e4db50cc5e7a3641fc6dc91f5a"}
[2019-11-12 15:03:15.109] DEBUG: IpfsHashAlreadyExists: Failed to apply transaction, because this IPFS hash is already registered on the blockchain.
    at IpfsTransactionHandler.throwIfCannotBeApplied (/home/vd/core/packages/core-transactions/dist/handlers/ipfs.js:40:19)
    at IpfsTransactionHandler.applyToSender (/home/vd/core/packages/core-transactions/dist/handlers/transaction.js:107:20)
    at IpfsTransactionHandler.applyToSender (/home/vd/core/packages/core-transactions/dist/handlers/ipfs.js:48:21)
    at IpfsTransactionHandler.apply (/home/vd/core/packages/core-transactions/dist/handlers/transaction.js:94:20)
    at WalletManager.applyTransaction (/home/vd/core/packages/core-state/dist/wallets/wallet-manager.js:297:34)
    at async WalletManager.applyBlock (/home/vd/core/packages/core-state/dist/wallets/wallet-manager.js:225:17)
    at async DatabaseService.applyBlock (/home/vd/core/packages/core-database/dist/database-service.js:58:9)
    at async AcceptBlockHandler.execute (/home/vd/core/packages/core-blockchain/dist/processor/handlers/accept-block-handler.js:9:13)
    at async Blockchain.processBlocks (/home/vd/core/packages/core-blockchain/dist/blockchain.js:308:33)
```

**Comment by dated** [comment]
I'll double check the exceptions added in #3216.

**Comment by dated** [comment]
Fixed in #3253 @vasild

**Pull request merged**